pkg/lightning/backend/local/region_job.go

```diff
@@ -975,14 +983,33 @@ func (q *regionJobRetryer) run(ctx context.Context) {
 			select {
 			case <-ctx.Done():
 				return
+			case <-q.done:
+				return
```

Comment:
use sub context? it duplicates with `case <-ctx.Done():`

Reply:
in current code, `run` and `close` are executed in two goroutines. If we use sub context, `close` need to wait `run` is called to initialize the sub context. The caller will be more complicated. I prefer current `done` channel

Reply:
we can init the context on `new`

Reply:
https://github.com/pingcap/tidb/pull/56112/commits/3f8f6c5b27ec0292fb156563e1be2a1e03287614